### PR TITLE
Refactor settings, improve build times

### PR DESCRIPTION
**Pull request by fiam**
- Rename clivalue_t and associated types/functions/macros to setting_t
- Use a build stamp file to avoid running the settings generator on every make invocation

### CODE REVIEW

**Comment by martinbudden** [comment]
This PR does two things as stated in it's description. It should be split into two. In particular I think the build stamp stuff can be merged now, but I'd like to further review the settings refactoring.

**Review by digitalentity — CHANGES_REQUESTED** [review]
@fiam please split thips PR into two - first one should be  makefile and generated settings optimization - I think this is what we need to have as soon as possible.

**Comment by fiam** [comment]
Although there are two changes in this PR, they are related and the second one builds on top of the first one. All the commits are logically separated in self-contained units so you can look at each commit separately. The only reason I put it all in just one PR is because things have been quite slow in the past week and I didn't want to submit half of these changes, wait until they were merged and then submit the rest of them. It seemed like I was saving everyone a bit of work.

Also, the settings refactoring is only renaming (just cosmetic changes) and moving one function from cli.c to settings.c, so we can expose it to other callers. Nothing major.

**Review by digitalentity — APPROVED** [review]

**Pull request merged**
